first_task.scala

```diff
@@ -0,0 +1,260 @@
+                    creationDate: Timestamp,
+                    score: Long,
+                    title: String,
+                    body: String)
+
+val questionsDs = spark.read
+      .option("multiLine", true)
+      .option("header", true)
+      .schema(Encoders.product[Question].schema)
+      .option("escape", "\"")
+      .csv("/FileStore/tables/Questions.csv")
+      .na.drop("all")
+      .as[Question]
+      
+questionsDs.cache()
+
+
+// COMMAND ----------
+
+case class Answer(id: Long,
+                  ownerUserId: Long,
+                  creationDate: Timestamp,
+                  partnerId: Long,
+                  score: Long,
+                  body: String)
+
+val answersDs = spark.read
+      .option("multiLine", true)
+      .option("header", true)
+      .schema(Encoders.product[Answer].schema)
+      .option("escape", "\"")
+      .csv("/FileStore/tables/Answers.csv")
+      .na.drop("all")
+      .as[Answer]
+
+answersDs.cache()
+
+// COMMAND ----------
+
+case class Tag(id: Long,
+               tag: String)
+
+val tagsDs = spark.read
+      .option("multiLine", true)
+      .option("header", true)
+      .schema(Encoders.product[Tag].schema)
+      .option("escape", "\"")
+      .csv("/FileStore/tables/Tags.csv")
+      .na.drop("all")
+      .as[Tag]
+
+tagsDs.cache()
+
+// COMMAND ----------
+
+// MAGIC %md
+// MAGIC Collect daily statistics (number of unique users posted, number of questions and answers) into a separate table and export to a Parquet file.
+
+// COMMAND ----------
+
+val questionNaFreeDs = questionsDs
+      .withColumn("creationDate", col("creationDate").cast(DateType))
+      .na.drop(Seq("creationDate"))
+val answersNaFreeDs = answersDs
+  .withColumn("creationDate", col("creationDate").cast(DateType))
+  .na.drop(Seq("creationDate"))
+
+val questionCols = Seq(
+  lit(null).alias("answerId"),
+  col("ownerUserId"),
+  col("creationDate").cast(DateType),
+  col("id").alias("questionId")
+)
+val answerCols = Seq(
+  col("id").alias("answerId"),
+  col("ownerUserId"),
+  col("creationDate").cast(DateType),
+  lit(null).alias("questionId")
+)
+val unionDf = questionNaFreeDs.select(questionCols: _*).unionAll(answersNaFreeDs.select(answerCols: _ *))
+val dailyStatisticsDf = unionDf
+  .groupBy("creationDate")
+  .agg(
+  countDistinct("ownerUserId").alias("unique_users"),
+  count("questionId").alias("number_of_questions"),
+  count("answerId").alias("number_of_answers")
+)
+
+dbutils.fs.rm("/FileStore/tables/daily.parquet", true) 
+dailyStatisticsDf.write.parquet("/FileStore/tables/daily.parquet")
+dailyStatisticsDf.show()
+
+// COMMAND ----------
+
+def countMeanAndMedianOfAnswerScores(dataset: Dataset[Answer], predicate: Answer => Boolean): (Double, Double) = {
+    val mean = dataset
+      .filter(predicate)
+      .select(avg("score"))
+      .take(1)(0)(0).toString().toDouble
+
+    val median = dataset.filter(predicate).stat.approxQuantile("score", Array(0.5), 0.1)(0)
+
+    (mean, median)
```

Comment:
also, for the sake of simplicity and single responsibility principle, I would split into 2 methods: one for median, one for mean.